src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,128 @@
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    "Can not create a connection to the DB", new Throwable());
+        }
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        String sql = "SELECT * FROM books WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(BookDaoImpl.setResult(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    "Can not create a connection to the DB", new Throwable());
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public Book update(Book book) {
+        String sql = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            statement.setLong(3, book.getId());
+            int executeUpdate = statement.executeUpdate();
+            if (executeUpdate < 1) {
+                throw new DataProcessingException(
+                        "Expected to update at leas one row, but updated 0 rows", new Throwable());
+            }
+
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t updated book " + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public boolean delete(Book book) {
+        boolean deleted;
+        String sql = "DELETE FROM books WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql)) {
+            statement.setLong(1, book.getId());
+            int executeUpdate = statement.executeUpdate();
+            deleted = executeUpdate >= 1;
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    "Can not create a connection to the DB", new Throwable());
+        }
+        return deleted;
+    }
+
+    private static Book setResult(ResultSet resultSet) {
+        try {
+            String title = resultSet.getString("title");
+            BigDecimal price = resultSet.getBigDecimal("price");
+            long key = resultSet.getObject("id", Long.class);
+            Book book = new Book();
+            book.setId(key);
+            book.setTitle(title);
+            book.setPrice(price);
+            return book;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Database access error or closed result set", e);
+        }
```

Comment:
```suggestion
```

Reply:
If there is an error, it will still be caught in the method where this method is called.

Reply:
not fixed